### PR TITLE
Update publishing article to remove out-of-date versions and clarify cross-plat

### LINKED ISSUE #31964: Misleading documentation regarding cross-platform framework-dependent publish
The documentation states the following:

> Publish an app **cross-platform** framework-dependent. A Linux 64-bit executable is created along with the dll file.
> `dotnet publish -r linux-x64 --self-contained false`

However, this command will not necessarily publish something that is **cross-platform**. It will not include the runtimes/\<RID\> folders and files that include platform-specific libraries and native assemblies that would be needed to run on platforms other than what is specified by the -r flag.

I think what this command is doing is a **platform-specific**, framework-dependent publish. It isn't only controlling the type of executable, but also controls which platform-specific dependencies are included.

Note that the document also claims:
> Apps published as framework-dependent are cross-platform

Which doesn't seem to always be true.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 85eba478-0374-bb9b-8fa7-d5cf7d520603
* Version Independent ID: 89967250-8c3b-1999-078f-94ad7518ed49
* Content: [Application publishing - .NET](https://learn.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained)
* Content Source: [docs/core/deploying/index.md](https://github.com/dotnet/docs/blob/main/docs/core/deploying/index.md)
* Product: **dotnet-fundamentals**
* GitHub Login: @adegeo
* Microsoft Alias: **adegeo**

---
[Associated WorkItem - 50930](https://dev.azure.com/msft-skilling/Content/_workitems/edit/50930)

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Remove old versions from the table. Considering the commands have been stable for the last 3 versions, there isn't a point in calling it out anymore.
- Clarified the example used (the issue that generated this PR)
- Clarified the descriptions of framework-dependent publishing when a platform is specified.

[Internal preview](https://review.learn.microsoft.com/en-us/dotnet/core/deploying/?branch=pr-en-us-33193)

Fixes #31964

### CODE REVIEW

File: docs/core/deploying/index.md [hunk 1]
```diff
@@ -147,7 +145,7 @@ dotnet publish -r win-x64
 
 ## Publish with ReadyToRun images
 
-Publishing with ReadyToRun images will improve the startup time of your application at the cost of increasing the size of your application. In order to publish with ReadyToRun see [ReadyToRun](ready-to-run.md) for more details.
+Publishing with ReadyToRun images will improve the startup time of your application at the cost of increasing the size of your application. For more information, see [ReadyToRun](ready-to-run.md).
```
Comment (gewarren) [inline docs/core/deploying/index.md:148]:
```suggestion
Publishing with ReadyToRun images improves the startup time of your application at the cost of increasing the size of your application. For more information, see [ReadyToRun](ready-to-run.md).
```

File: docs/core/deploying/index.md [hunk 2]
```diff
@@ -82,20 +85,15 @@ Your app can run only if the version of .NET your app targets is already install
...
 
-Publish an app cross-platform framework-dependent. An executable that targets your current platform is created along with the *dll* file.
+Publish an app cross-platform framework-dependent. An executable that targets your current platform is created along with the *dll* file. Any platform-specific dependencies are published with the app.
 
 ```dotnetcli
 dotnet publish
 ```
 
-Publish an app cross-platform framework-dependent. A Linux 64-bit executable is created along with the *dll* file. This command doesn't work with .NET Core SDK 2.1.
+Publish an app platform-specific framework-dependent. A Linux 64-bit executable is created along with the *dll* file. Only the targeted platform's dependencies are published with the app.
```
Comment (gewarren) [inline docs/core/deploying/index.md:96]:
```suggestion
Publish an app as platform-specific and framework-dependent. A Linux 64-bit executable is created along with the *dll* file. Only the targeted platform's dependencies are published with the app.
```

File: docs/core/deploying/index.md [hunk 3]
```diff
@@ -82,20 +85,15 @@ Your app can run only if the version of .NET your app targets is already install
...
 
-The following disadvantage only applies to .NET Core 2.1 SDK.
-
-- **Use the `dotnet` command to start the app**\
-Users must run the `dotnet <filename.dll>` command to start your app. .NET Core 2.1 SDK doesn't produce platform-specific executables for apps published framework-dependent.
-
 ### Examples
 
-Publish an app cross-platform framework-dependent. An executable that targets your current platform is created along with the *dll* file.
+Publish an app cross-platform framework-dependent. An executable that targets your current platform is created along with the *dll* file. Any platform-specific dependencies are published with the app.
```
Comment (gewarren) [inline docs/core/deploying/index.md:90]:
```suggestion
Publish an app as cross-platform and framework-dependent. An executable that targets your current platform is created along with the *dll* file. Any platform-specific dependencies are published with the app.
```

File: docs/core/deploying/index.md [hunk 4]
```diff
@@ -46,22 +46,25 @@ Cross-platform binaries can be run on any operating system as long as the target
...
 
-You can create an executable for a specific platform by passing the `-r <RID> --self-contained false` parameters to the [`dotnet publish`](../tools/dotnet-publish.md) command. When the `-r` parameter is omitted, an executable is created for your current platform. Any NuGet packages that have platform-specific dependencies for the targeted platform are copied to the publish folder. If you don't need a platform-specific executable, you can specify `<UseAppHost>False</UseAppHost>` in the project file. For more information, see [MSBuild reference for .NET SDK projects](../project-sdk/msbuild-props.md#useapphost).
+You can publish a framework-dependent app that is platform-specific by passing the `-r <RID> --self-contained false` parameters to the [`dotnet publish`](../tools/dotnet-publish.md) command. Publishing in this way is the same as [publish framework-dependent](#publish-framework-dependent), except that platform-specific dependencies are handled differently. If the app uses a NuGet package that has platform-specific implementations, only the targeted platform's dependencies are copied. These dependencies are copied directly to the *publish* folder.
```
Comment (gewarren) [inline docs/core/deploying/index.md:63]:
```suggestion
You can publish a framework-dependent app that's platform-specific by passing the `-r <RID> --self-contained false` parameters to the [`dotnet publish`](../tools/dotnet-publish.md) command. Publishing in this way is the same as [publish framework-dependent](#publish-framework-dependent), except that platform-specific dependencies are handled differently. If the app uses a NuGet package that has platform-specific implementations, only the targeted platform's dependencies are copied. These dependencies are copied directly to the *publish* folder.
```

File: docs/core/deploying/index.md [hunk 5]
```diff
@@ -46,22 +46,25 @@ Cross-platform binaries can be run on any operating system as long as the target
...
+
+While technically the binary produced is cross-platform, by targeting a specific platform, your app isn't guaranteed to run cross-platform. So you can run `dotnet <filename.dll>`, but the app may crash once it tries to access platform-specific dependencies that are missing.
```
Comment (gewarren) [inline docs/core/deploying/index.md:65]:
```suggestion
While technically the binary produced is cross-platform, by targeting a specific platform, your app isn't guaranteed to run cross-platform. You can run `dotnet <filename.dll>`, but the app may crash when it tries to access platform-specific dependencies that are missing.
```

File: docs/core/deploying/index.md [hunk 6]
```diff
@@ -46,22 +46,25 @@ Cross-platform binaries can be run on any operating system as long as the target
...
 
-> [!IMPORTANT]
-> .NET SDK 2.1 doesn't produce platform-specific executables when you publish an app framework-dependent.
+The cross-platform binary of your app can be run with the `dotnet <filename.dll>` command, and can be run on any platform.
 
-The cross-platform binary of your app can be run with the `dotnet <filename.dll>` command, and can be run on any platform. If the app uses a NuGet package that has platform-specific implementations, all platforms' dependencies are copied to the publish folder along with the app.
+### Platform-specific framework-dependent
```
Comment (gewarren) [inline docs/core/deploying/index.md:61]:
```suggestion
### Platform-specific and framework-dependent
```

File: docs/core/deploying/index.md [hunk 7]
```diff
@@ -46,22 +46,25 @@ Cross-platform binaries can be run on any operating system as long as the target
...
+Publishing an app as framework-dependent produces a [cross-platform binary](#produce-a-cross-platform-binary) as a *dll* file, and a [platform-specific executable](#produce-an-executable) that targets your current platform. The *dll* is cross-platform while the executable isn't. For example, if you publish an app named **word_reader** and target Windows, a *word_reader.exe* executable is created along with *word_reader.dll*. When targeting Linux or macOS, a *word_reader* executable is created along with *word_reader.dll*. If the app uses a NuGet package that has platform-specific implementations, all platforms' dependencies are copied to the *publish\\runtimes\\{platform}* folder.
```
Comment (gewarren) [inline docs/core/deploying/index.md:57]:
```suggestion
Publishing an app as framework-dependent produces a [cross-platform binary](#produce-a-cross-platform-binary) as a *dll* file, and a [platform-specific executable](#produce-an-executable) that targets your current platform. The *dll* is cross-platform while the executable isn't. For example, if you publish an app named **word_reader** and target Windows, a *word_reader.exe* executable is created along with *word_reader.dll*. When targeting Linux or macOS, a *word_reader* executable is created along with *word_reader.dll*. If the app uses a NuGet package that has platform-specific implementations, dependencies from all platforms are copied to the *publish\\runtimes\\{platform}* folder.
```

Comment (gewarren) [reply docs/core/deploying/index.md:57]: Just reads a little easier.

File: docs/core/deploying/index.md [hunk 8]
```diff
@@ -17,26 +17,26 @@ When an executable is produced, you can specify the target platform with a runti
...
 
 ## Produce an executable
 
-Executables aren't cross-platform. They're specific to an operating system and CPU architecture. When publishing your app and creating an executable, you can publish the app as [self-contained](#publish-self-contained) or [framework-dependent](#publish-framework-dependent). Publishing an app as self-contained includes the .NET runtime with the app, and users of the app don't have to worry about installing .NET before running the app. Apps published as framework-dependent don't include the .NET runtime and libraries; only the app and 3rd-party dependencies are included.
+Executables aren't cross-platform, they're specific to an operating system and CPU architecture. When publishing your app and creating an executable, you can publish the app as [self-contained](#publish-self-contained) or [framework-dependent](#publish-framework-dependent). Publishing an app as self-contained includes the .NET runtime with the app, and users of the app don't have to worry about installing .NET before running the app. Publishing an app as framework-dependent doesn't include the .NET runtime; only the app and 3rd-party dependencies are included.
```
Comment (gewarren) [inline docs/core/deploying/index.md:31]: Acrolinx wouldn't like "publishing an app" but if you prefer it it's fine with me.

Comment (gewarren) [inline docs/core/deploying/index.md:31]:
```suggestion
Executables aren't cross-platform, they're specific to an operating system and CPU architecture. When publishing your app and creating an executable, you can publish the app as [self-contained](#publish-self-contained) or [framework-dependent](#publish-framework-dependent). Publishing an app as self-contained includes the .NET runtime with the app, and users of the app don't have to worry about installing .NET before running the app. Publishing an app as framework-dependent doesn't include the .NET runtime; only the app and third-party dependencies are included.
```

File: docs/core/deploying/index.md [hunk 9]
```diff
@@ -17,26 +17,26 @@ When an executable is produced, you can specify the target platform with a runti
 
 The following table outlines the commands used to publish an app as framework-dependent or self-contained, per SDK version:
```
Comment (gewarren) [inline docs/core/deploying/index.md:18]:
```suggestion
The following table outlines the commands used to publish an app as framework-dependent or self-contained:
```

File: docs/core/deploying/index.md [hunk 10]
```diff
@@ -46,22 +46,25 @@ Cross-platform binaries can be run on any operating system as long as the target
...
 The following command produces a cross-platform binary:
 
-| Type                                                                                 | SDK 2.1 | SDK 3.x | SDK 5.0 | SDK 6.0 | Command |
-| -----------------------------------------------------------------------------------  | ------- | ------- | ------- | ------- | ------- |
-| [framework-dependent cross-platform binary](#publish-framework-dependent).           | ✔️      | ✔️      | ✔️      | ✔️      | [`dotnet publish`](../tools/dotnet-publish.md) |
+| Type                                                                                | Command |
+| ----------------------------------------------------------------------------------- | ------- |
+| [framework-dependent cross-platform binary](#publish-framework-dependent).          | [`dotnet publish`](../tools/dotnet-publish.md) |
 
 ## Publish framework-dependent
```
Comment (gewarren) [inline docs/core/deploying/index.md:53]:
```suggestion
## Framework-dependent
```

Comment (gewarren) [reply docs/core/deploying/index.md:53]: To be consistent with the heading on line 61.

**Review by gewarren — APPROVED** [review]
Left some suggestions for consideration.

File: docs/core/deploying/index.md (hunk 10 above)
Comment (adegeo) [reply docs/core/deploying/index.md:53]: The section below (unedited in the PR) is `Publish self-contained`. The section on line 61 is a subsection of this one, describing the behavior of "platform-specific" feature of framework-dependent.

**Pull request merged**